### PR TITLE
fix(kubescape): stop masking the baked policy library when downloads are off

### LINKED ISSUE #874: kubescapeOffline=enable "failed to list frameworks","error":"open /home/nonroot/.kubescape/security.json: no such file or directory
# Description
I am trying to deploy kubescape-operator in offline mode with the following values:

```
certificates:
  strategy: initContainer
capabilities:
  kubescapeOffline: enable
kubescape:
  downloadArtifacts: false
  skipUpdateCheck: true
grypeOfflineDB:
  enabled: true
  image:
    tag: v6-latest
```

At first, I stumbled upon #857. Therefore I now deploy directly from the main branch.

# Environment
OS: Ubuntu 24.04 LTS
Chart version: kubescape-operator 1.40.2 with recent commits in main branch
Kubernetes: v1.35
Installation method: Helm

# Steps To Reproduce
1. Install kubescape-operator chart from main branch with the values shown above
2. Trigger a scan: `kubectl create job --from=cronjob/kubescape-scheduler test-scan -n kubescape`
3. Wait for completion and check logs: `kubectl logs deployment/kubescape -n kubescape`
4. Observe the error 
```
{"level":"error","ts":"2026-07-07T09:03:12Z","msg":"scanning failed","ID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd","error":"failed to scan. reason: 'open /home/nonroot/.kubescape/security.json: no such file or directory'"}
```
5. Verify: `kubectl get workloadconfigurationscans -A` → no resources

# Expected behavior
With `keepLocal: true` and/or `kubescapeOffline: enable`, scans should be executed and results persisted.

# Actual Behavior
Scan fails with:
```
{"level":"info","ts":"2026-07-07T08:58:47Z","msg":"Starting Kubescape server","version":"4.0.8","commit":"d7539c2264560a8685f59e89a731d6de833258a6","date":"2026-05-08T16:49:49Z"}
{"level":"info","ts":"2026-07-07T08:58:47Z","msg":"credentials loaded from path","path":"/etc/credentials","accessKeyLength":0,"accountLength":0}
{"level":"info","ts":"2026-07-07T08:58:47Z","msg":"Starting swagger UI. baseURI: /openapi/v2/, docsEP: docs, rapidocEP: rapi, swaggerui: swaggerui"}
{"level":"info","ts":"2026-07-07T08:58:47Z","msg":"Started Kubescape server","port":"8080","version":"4.0.8"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"REST API received scan request","scanID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd","format":""}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"requesting scan","scanID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd","api":"v1/scan"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"triggering scan","scanID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"scan triggered","ID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"Kubescape scanner initializing..."}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"Using CRD-based host sensor data collection (no deployment needed)"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"node-agent status: Successfully verified CRD access","osReleaseFiles":1}
{"level":"warn","ts":"2026-07-07T09:03:12Z","msg":"failed to list frameworks","error":"open /home/nonroot/.kubescape/security.json: no such file or directory"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"Initialized scanner"}
{"level":"info","ts":"2026-07-07T09:03:12Z","msg":"Loading policies..."}
{"level":"error","ts":"2026-07-07T09:03:12Z","msg":"scanning failed","ID":"2debd830-ba2e-4c06-ba71-b8634fb5bbbd","error":"failed to scan. reason: 'open /home/nonroot/.kubescape/security.json: no such file or directory'"}
```

`kubectl get workloadconfigurationscans -A` returns no resources.

### PR DESCRIPTION
**Pull request by ahauserv**
## Overview
Fixes `kubescape.downloadArtifacts: false`, which does not work today.
The image bakes the policy library into `/home/nonroot/.kubescape` (17 files, `nsa.json`, `security.json`, ...)
The chart mounts a writable `emptyDir` over that directory, so with downloads off there is nothing left to fall back to.

Nothing about the deployment looks wrong; only the scan fails:

* `failed to list frameworks … open /home/nonroot/.kubescape/security.json: no such file or directory` (#874)
* with no frameworks named: `unknown policy kind`
* with them named: `framework: nsa: framework from file not matching`
* each preceded by `failed to get policies from github release, loading policies from cache`, the fetch an air-gapped install cannot make

This makes the mount and the volume conditional on `downloadArtifacts`, so the baked library stays visible.
With downloads on (the default) nothing changes.

`readOnlyRootFilesystem: true` is why the volume exists at all: the scanner needs somewhere writable to download into.
With downloads off it writes nothing there.

## How to Test
```
helm template test charts/kubescape-operator --set clusterName=x | grep -A2 'name: kubescape-volume'
helm template test charts/kubescape-operator --set clusterName=x --set kubescape.downloadArtifacts=false | grep -A2 'name: kubescape-volume'
```
The first prints the mount and the volume. The second prints nothing.

Isolated to that one mount, using two containers from the same image with the same environment and a read-only root filesystem: without the mount a scan completes; with an empty volume over `~/.kubescape` it fails with `unknown policy kind`.

Confirmed in a cluster on a read-only root filesystem: the scan completes and persists its results, so nothing needs writing to ~/.kubescape when there is nothing to download.

## Related issues/PRs:

* Resolves #874
* Completes #857 — mounting the parent directory fixed the `config.json` crash and masked the
  baked library; this keeps the library and drops the mount only when it is not needed

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **New Features**
  * Added clearer documentation for the artifact download setting, including how bundled policies behave when downloads are disabled.
  * Downloads can now be disabled while retaining the policy library included in the container image.

* **Bug Fixes**
  * Prevented unnecessary writable policy storage from being mounted when artifact downloads are disabled.
  * Preserved host-scanner configuration mounting in this mode.

### CODE REVIEW

File: charts/kubescape-operator/templates/kubescape/deployment.yaml [hunk 1]
```diff
@@ -240,9 +240,11 @@ spec:
         - name: {{ $components.cloudSecret.name }}
           mountPath: /etc/credentials
           readOnly: true
+        {{- if .Values.kubescape.downloadArtifacts }}
         - name: kubescape-volume
           mountPath: /home/nonroot/.kubescape
           subPath: config.json
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (ahauserv) [reply charts/kubescape-operator/templates/kubescape/deployment.yaml:247]: `kubescape-volume` is an empty `emptyDir`, it never supplied a `config.json`. the `subPath: config.json` is a dead leftover from #301 and the line should be deleted in a follow up PR.

**Review by matthyx — APPROVED** [review]
Verified the fix — rendered both branches with `helm template` and ran the chart's `helm unittest` suite (`kubescape_download_artifacts_test.yaml` + full suite, 53/53 tests, 996/996 snapshots pass):

- `downloadArtifacts: true` (default): `kubescape-volume` mount/emptyDir present, unchanged.
- `downloadArtifacts: false`: `kubescape-volume` mount/emptyDir is dropped, so `/home/nonroot/.kubescape` keeps the image's baked policy library, matching the PR description and resolving #874. `host-scanner-definition` stays mounted in both cases.

One combination worth checking before merge, matching CodeRabbit's inline note on `deployment.yaml:243-247`: with `capabilities.kubescapeOffline: enable` **and** `kubescape.downloadArtifacts: false` together, this change now also removes the writable `kubescape-volume`/`config.json` mount that offline mode previously always got (existing `offline_mode_test.yaml` only covers offline mode with the default `downloadArtifacts: true`, so this combo isn't under test). `kubescape`'s own `ClusterConfig`/`LocalConfig.UpdateCachedConfig()` persists to `~/.kubescape/config.json`, and with `readOnlyRootFilesystem: true` and no writable mount there, any code path that calls it (e.g. `kubescape config set`, `GenerateAccountID`) would fail to write. I couldn't confirm the `ksserver` request path used by this chart calls that write during normal scans, so this may be a non-issue in practice — but it'd be safer to gate the mount on `downloadArtifacts || kubescapeOffline` (or add a test for the combined case) so a future change doesn't silently break it.

This is a narrow edge case, not a regression in the described fix, and everything else checks out (renders cleanly, tests green, CI green). Approving — please just confirm/track the offline+no-download combination above.
